mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -90,7 +90,12 @@ private[clustering] trait KMeansParams extends Params with HasMaxIter with HasFe
    * @return output schema
    */
   protected def validateAndTransformSchema(schema: StructType): StructType = {
-    SchemaUtils.checkColumnType(schema, $(featuresCol), new VectorUDT)
+    val typeCandidates = List( new VectorUDT,
+      new ArrayType(DoubleType, true),
```

Comment:
Thinking about this, let's actually disallow nullable columns.  KMeans won't handle nulls properly.

Reply:
Also, IntelliJ may warn you about passing boolean arguments as named arguments; that'd be nice to fix here.